test/src/k-provider/ovp/regex-action-handler.spec.js

```diff
@@ -1,22 +1,35 @@
-import {finalMediaConfigAllSourcesModified, mediaConfig, responseDataFromBE} from './regex-action-handler-data';
+import {
+  finalMediaConfigAllSourcesModified,
+  finalMediaConfigPlayManifestSourcesModified,
+  mediaConfig,
+  responseDataFromBE
+} from './regex-action-handler-data';
 import RegexActionHandler from '../../../../src/k-provider/ovp/regex-action-handler';
 import OVPMediaEntryLoader from '../../../../src/k-provider/ovp/loaders/media-entry-loader';
 import OVPConfiguration from '../../../../src/k-provider/ovp/config';
 
 describe('handleRegexAction', function () {
   let data = new Map();
-  let mediaEntryLoader;
-  let mediaConfigForTest;
+  let mediaEntryLoader, mediaConfigForTest;
 
   before(() => {
     mediaEntryLoader = new OVPMediaEntryLoader({entryId: 'a', ks: 'a', redirectFromEntryId: true});
     mediaEntryLoader.response = responseDataFromBE;
     data.set('media', mediaEntryLoader);
   });
 
+  afterEach(() => {
+    RegexActionHandler._pingECDNUrl.restore();
+  });
+
   it('should modify all URLs', done => {
+    sinon.stub(RegexActionHandler, '_pingECDNUrl').callsFake(function () {
+      return new Promise(resolve => {
+        resolve(true);
+      });
+    });
     OVPConfiguration.set({replaceHostOnlyManifestUrls: false});
-    mediaConfigForTest = {...mediaConfig};
+    mediaConfigForTest = JSON.parse(JSON.stringify({...mediaConfig}));
```

Comment:
why do you need stringify + parse ? is the object you're getting different than {...mediaConfig} ?

Reply:
@SivanA-Kaltura  the function that replaces the host in the urls is modifying the original variable (by reference), this is affecting the following tests, since the first test is changing the values of the original mediaConfig. for now i decided to deep clone the original object in each test to not affect one another.

Reply:
why not reset the original variable in afterEach ?

Reply:
hmmm.. even if I will reset only the sources object i would still have to use stringify + parse to deep clone that part..